File: c#Tools/searching_algorithms.cs
namespace searching_algorithms {
    class Searching_Algorithms {
        public static bool LinearSearch(int[] array, int toSearch) {
            foreach (int item in array) {
                if (item == toSearch) {
                    return true;
                }
            }

            return false;
        }

        public static bool BinarySearch(int[] array, int toSearch) {
            int leftPointer = 0;
            int rightPointer = array.Length - 1;
            int midPointer;

            while (rightPointer >= leftPointer) {
                midPointer = (leftPointer + rightPointer) / 2;

                if (array[midPointer] == toSearch) {
                    return true;
                }
                else if (array[midPointer] < toSearch) {
                    leftPointer = midPointer + 1;
                }
                else {
                    rightPointer = midPointer - 1;
                }
            }

            return false;
        }
    }
}

File: jsTools/searchingAlgorithms.js
function linearSearch(array, toSearch) {
    for (i = 0; i < array.length; i++) {
        if (array[i] == toSearch) {
            return true;
        }
    }
    return false;
}

function binarySearch(array, toSearch) {
    leftPointer = 0;
    rightPointer = array.length - 1;

    while (rightPointer >= leftPointer) {
        midPointer = Math.floor((leftPointer + rightPointer) / 2);

        if (array[midPointer] == toSearch) {
            return true;
        }
        else if (array[midPointer] < toSearch) {
            leftPointer = midPointer + 1;
        }
        else {
            rightPointer = midPointer - 1;
        }
    }

    return false;
}

arr = [1, 2, 5, 7, 12, 64, 560, 1245, 124526];

console.log(binarySearch(arr, 1));
console.log(binarySearch(arr, 3));
console.log(binarySearch(arr, 12));
console.log(binarySearch(arr, 124526));